provider/provider.go

```diff
@@ -110,7 +110,6 @@ func (p *dockerNativeProvider) Check(ctx context.Context, req *rpc.CheckRequest)
 	inputs, err := plugin.UnmarshalProperties(req.GetNews(), plugin.MarshalOptions{
 		KeepUnknowns: false,
 		SkipNulls:    true,
-		KeepSecrets:  true,
```

Comment:
It seems like its worth refactoring `plugin.UnmarshalProperties(req.GetNews(), plugin.MarshalOptions{ KeepUnknowns: false, SkipNulls: true })` into `p.unmarshalProperties(req.GetNews())`. with `p.unmarshalProperties` a wrapper function that unifies `MarshalOptions`.

Reply:
We create `MarshalOptions` multiple times with largely the same arguments, which we define each time. To keep DRY, we could move the `plugin.UnmarshalProperties` call into a function that itself has the defaults, preventing us from needing to provide them each time at the call-site. 

This is what I did in the go provider SDK:
https://github.com/pulumi/pulumi-go-provider/blob/e8f254577aee9fba48c3a8cf7238ef95a4b14141/provider.go#L584-L599